### ISSUE TITLE
screen reader accessibility roadmap

### ISSUE THREAD

**Issue by PlasticMemmories** [gkasdorf/memmy#192]
# memmy screen reader accessibility roadmap

I'm not very familiar with react native development, but this [document on accessibility](https://reactnative.dev/docs/accessibility) may help

## homepage

* there is an unlabeled element at the top of the home page that when clicked, takes you to the subscriptions page. This should be labeled with a descriptive title, in this case, that label would be "subscriptions"
* there are 2 more unlabeled elements. One takes you to the sort options, and the other takes you to a menu that has options such as "all", "local", and "subscribed". As above, these elements should have a descriptive label that lets a screen reader user understand what the button does
* the post information you see on each post in the list should be modified, currently, the information  sounds like this. "post title, 1h 20 0 0 community", which doesn't help a screen reader user. What are these numbers? What do they mean? My recommendation would be to add what that number is counting, for example if one of those numbers is upvotes and another is comments, it should sound like (number)upvotes, (number)comments. Another recommendation would be to add the words "in community" before the community name, so the entire thing would sound something like this. "Post, 1hour ago, 20 upvotes, 0 comments, in community community".
* there should be an accessibility action that lets a user navigate to the profile of the author who posted a post, as well as one that lets the user go to the community that the post is in. Accessibility actions are talked about in the document I linked above
* currently, as a screen reader user, there seems to be no way to vote on a post, consider adding accessibility actions for that as well

## posts

* the bookmark, reply, and share buttons are not labeled
* as mentioned in the homepage section, there are a bunch of random numbers that aren't explained to the user, are they upvotes? comments left? the user won't know
* it would be good to also get the same accessibility actions as mentioned in the homepage section even when a post is open.

## comments.

* currently, when reading a comment, if there are multiple lines in a comment, the linebreaks aren't noticed, so instead of being able to swipe right to the next line and left for the previous line in a comment, you get a large bit of text read out by the screen reader, which isn't good if a user is reading a rather long comment that is split up into multiple lines.
* currently, as mentioned in the posts section, there is no way to vote, on a comment with a screen reader. Accessibility actions to vote would be beneficial here
* an accessibility action that would let you go to the profile of the commenter would also be useful, as well as an action that lets you reply to the comment
* currently, a screen reader user can't keep track of the conversation because there is no way to know the depth of the comment in a thread, or who is replying to who. Either one of these pieces of information would be beneficial
* there are more numbers that need to be explained as mentioned above in the posts and homepage sections

## profiles page

* the settings button on top of the profile page isn't labeled
* again, more unexplained numbers
